src/boot/efi/sbat.c.in

```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+/*
+ * Copyright © 2021 Canonical Ltd
+ *
+ */
+
+static const char sbat[]
+__attribute__((section (".sbat")))
+__attribute__((__used__))
+__attribute__((aligned(512))) = ""
+"sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md\n"
+"@sbat_project@,1,systemd EFI boot,@sbat_project@,@PROJECT_VERSION@,@PROJECT_URL@\n"
+"@sbat_project@.@sbat_distro@,@sbat_distro_generation@,@sbat_distro_summary@,@sbat_distro_pkgname@,@sbat_distro_version@,@sbat_distro_url@";
```

Comment:
this just concats a bunch of strings, no? i.e. you can just pass te fields in as macros (one of this is actually a macro anyway).

i.e. something like:

```c
#ifdef SBAT_PROJECT
static const char sbat[] _section_(".sbat") _used_ _align_(512) =
        "sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md\n"
        SBAT_PROJECT ",1,systemd EFI boot," SBAT_PROJECT "," STRINGIFY(PROJECT_VERION) "," PROJECT_URL "\n"
…;
#endif
```

i.e. for simple things like this, regular C macros should totally suffice. I wouldn't generate any code for this. And thing should become a lot simpler.

Reply:
(and let's really hope noone embedds a comma in any of these fields... how is that supposed to be escaped even?

Reply:
> this just concats a bunch of strings, no? i.e. you can just pass te fields in as macros (one of this is actually a macro anyway).

i can do that too, but then i think i'd move all the processing of the options to the top level meson.build to just be part of the standard config.h.

Reply:
> (and let's really hope noone embedds a comma in any of these fields... how is that supposed to be escaped even?

there is quoting and escaping defined, but if there are mistakes like that shim will refuse to validate sbat and fail booting it. So the error scenario for invalid things is quite dramatic. More validation can be done on the input strings, but it's mostly ok as is.